mysql/resource_user.go

```diff
@@ -198,6 +212,14 @@ func CreateUser(ctx context.Context, d *schema.ResourceData, meta interface{}) d
 		}
 	}
 
+	retainPassword := d.Get("retain_old_password").(bool)
+	if retainPassword {
+		_, err := checkRetainCurrentPasswordSupport(ctx, meta)
+		if err != nil {
+			return diag.Errorf("%v", err)
```

Comment:
Please make error message more descriptive by saying, where it came from.
Go usually goes with "stacktrace-like" errors, where more and more errors are prepended to the error message. That makes debugging easier.